src/utils/filesystem.cpp

```diff
@@ -843,6 +844,43 @@ bool verify_file(const std::string &fname,
     return true;
 }
 
+bool verify_file_size(const std::string &fname, const int64_t &expected_fsize)
```

Comment:
I suggest that you can refactor function `verify_file` to decrease similar code in your next pull request.